R/GGIR.R

```diff
@@ -151,11 +158,20 @@ GGIR = function(mode = 1:5, datadir = c(), outputdir = c(),
   }
   if (dopart1 == TRUE) {
     if (verbose == TRUE) print_console_header("Part 1")
-    g.part1(datadir = datadir, outputdir = outputdir, f0 = f0, f1 = f1,
-            studyname = studyname, myfun = myfun,
-            params_rawdata = params_rawdata, params_metrics = params_metrics,
-            params_cleaning = params_cleaning, params_general = params_general,
-            verbose = verbose)
+    if (params_general[["dataFormat"]] == "raw") {
+      g.part1(datadir = datadir, outputdir = outputdir, f0 = f0, f1 = f1,
+              studyname = studyname, myfun = myfun,
+              params_rawdata = params_rawdata, params_metrics = params_metrics,
+              params_cleaning = params_cleaning, params_general = params_general,
+              verbose = verbose)
+    } else {
+      # Skip g.part1, but instead convert epoch data to a format that
+      # looks as if it came out of g.part1
+      convertEpochData(datadir = datadir,
```

Comment:
Should we inform the user with a message in the console if part 1 is skipped? In that way they are informed that the configurations that they select for part 1 do not have any effect on the data processing (because data are not raw format).

Reply:
To do:
- [ ] Reflect on how we can make extraction of epochsize easier and avoid on relying input from user.
- [ ] tz is still needed.
- [ ] Reflect on whether we can send warning to user when they provide non-default GGIR part 1 arguments.

Reply:
After reflection, I have now decided to show the warning message. It is not the most elegant solution but any other solution I can think of would require a lot of work. For example, we would have to label the loaded metrics as "externally derived metrics" and communicate that in all the GGIR output.